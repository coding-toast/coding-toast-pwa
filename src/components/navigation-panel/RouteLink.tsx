import * as React from 'react';
import { Nav } from 'react-bootstrap';
import Link from 'next/link';

interface IRouteLinkProps {
  href: string;
  label: string;
}

const RouteLink: React.FC<IRouteLinkProps> = (props) => (
  <Nav.Item>
    <Nav.Link>
      <Link href={props.href} passHref>
        <a className='text-text-color-primary'>{props.label}</a>
      </Link>
    </Nav.Link>
  </Nav.Item>
);

export default RouteLink;
