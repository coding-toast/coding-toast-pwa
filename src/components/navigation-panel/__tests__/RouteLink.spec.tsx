import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RouteLink from '../RouteLink';

describe('<RouteLink />', () => {
  it('renders correctly', () => {
    const component = shallow(<RouteLink href='/hello_world' label='Hello World!' />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
