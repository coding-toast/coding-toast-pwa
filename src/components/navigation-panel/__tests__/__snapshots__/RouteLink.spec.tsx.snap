// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RouteLink /> renders correctly 1`] = `
<NavItem>
  <NavLink
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "SafeAnchor",
        "render": [Function],
      }
    }
    disabled={false}
  >
    <Link
      href="/hello_world"
      passHref={true}
    >
      <a
        className="text-text-color-primary"
      >
        Hello World!
      </a>
    </Link>
  </NavLink>
</NavItem>
`;
